javascript-review.js

```diff
@@ -3,112 +3,203 @@
+const objectOne = {
+  numVariable: 2,
+  strVariable: "haha!",
+  arrVariable: [1, 2, 3, 4],
+  display() {
+    console.log(this.numVariable);
+  }
+}
 
 // 2.
 
-// Write a constructor function (to create objects) that takes one argument. It should use that argument to set the value of the first attribute of the object. There should be a second attribute that stores the value ‘red’. 
+// Write a constructor function (to create objects) that takes one argument. It should use that argument to set the value of the first attribute of the object. There should be a second attribute that stores the value ‘red’.
+
+const fruitConstructor = (name) => {
+  return {
+    name: name,
+    color: "red"
+  }
+}
 
 // Note that a constructor function is the old school syntax that was used in javascript before the sugar class syntax came along. 
 
 // 3.
 
 // Can you access variables defined outside of functions within the scope of functions in JS? Show a simple example of this being used? (Define a constant in global scope and use that in a very simple function.)
 
+//Yes, we can access it. 
+const counter = 1;
+const displayCounter = () => {
+  console.log(counter);
+}
+
 // 4.
 
 // What two ways can be used to access the properties of objects in JS? Give examples of both (although assume the objects have been defined).
+console.log(objectOne.arrVariable);
+console.log(objectOne["arrVariable"]);
+
 
 // 5.
 
 // What is JSON? Be clear but concise.
+// It's a popular and lightweighted way of storing information that's quite human-readable and looks like javascript object.
+
 
 // 6.
 
 // What is a callback?
+// function that's being passed into a function to be used in that function.
+
 
 // 7.
 
 // Define a function with that takes two number arguments and a callback. The function will add the two numbers and pass them to the callback as an argument. Now call this function and in the callback simply console.log out the argument that has been passed through.
+const sum = (num1, num2, displayCallback) => {
+  displayCallback(num1 + num2)
+  return num1 + num2
+}
+const display = (result) => {
+  console.log(result);
+}
+
+sum(1, 2, display);
 
 // 8.
 
 // Why do we use promises?
+// because it allows us to run our code asycronously and our application won't be bottleneck-ed by a slow internet speed or huge background processing.
 
 // 9.
 
 // What are the two methods at our disposal if we have hold of a promise? What are the names of the functions that are related to these functions (that are called when the promise is fulfilled or fails).
+//then & await(given that the function is declared as async function).
 
 // 10.
 
 // What is the DOM?
+// the HTML file that being rendered. It provides a library of function that allows developer to select element from the document and manipulate the document itself. 
```

Comment:
not just the html file, everything that's in the window of the browser, even cookies is part of the DOM